Adaptation for AHK 2.0, optimization for non-standard layouts and error correction

Ulyanov-programmer opened the pull request:
I wanted to add better keyboard shortcuts for switching, but...

- Added compatibility with third-party keyboard layouts such as Dvorak, Zubachev layout and qphyx. This was achieved by using **SC-codes** instead of strict binding to letters.
- The code has been optimized (_I understand that the script was written a long time ago, but it was terrible_) and reduced by about 20%.
- Adapted for AHK 2.0
- The notification window has been improved, and the Windows tray is now used.
- Incorrectly working keyboard shortcuts have been redesigned to enable **normal WASD** (`<+Space`) and **normal VIM** (`>+Space`) mode.
- The hotkeys _not listed_ in the README are disabled (moreover, it was not clear to me why they were needed at all).

Ulyanov-programmer committed:
Partially adapted to AHK 2.0, improved formatting.

Ulyanov-programmer committed:
Work on adaptation for 2.0 and adaptation for layouts has been completed.

Optimized functions and disabled broken keyboard shortcuts.

Ulyanov-programmer committed:
Fixes and improvements.

The information window is made beautiful.
Fixed an issue where the keyboard shortcuts for fast scrolling were not working.

Ulyanov-programmer committed:
The commented code was returned.

4strid commented:
right on ! lol , "terrible" is , generous , to be frank . This was my first and only project using AHK and I , did not bother learning the language before making it . . . Lemme take it for a spin real quick but , very likely this will be version 2.0 of mouse-control :>

Ulyanov-programmer committed:
Fixed: the hint display function to fire indefinitely.

Ulyanov-programmer committed:
Added a function to activate actions when double-tapping.

The scroll keys have added functionality when pressed twice.

The keyboard shortcuts that were used for scrolling several times have been removed.

Ulyanov-programmer committed:
Added emulation of holding the mouse key.

Ulyanov-programmer committed:
Input mode variables have been abrogated.

Increased the smoothness of mouse input.

Ulyanov-programmer committed:
Fixed a bug where when exiting quick mode, the return was only in VIM.

Ulyanov-programmer committed:
The hotkey responsible for the Yank function is enabled.

Ulyanov-programmer commented:
Hello! I'm afraid to sound intrusive, but how is your progress?

4strid commented:
i sent you an email just now . i don't typically have access to a computer at this time in my life heh 😅.......

 but i'm looking at it on a friend's laptop now :>

Ulyanov-programmer committed:
Commented keyboard shortcuts have been removed.

4strid commented:
ok , if , i'm , completely honest , this is taking longer to look at than i really have available to me . I think I'd like to hand the reigns of the project off to you , entirely , if that works for you ?

Ulyanov-programmer commented:
The worst decisions are made very quickly. If you can wait a couple more days to make a decision, it will be for the best.

4strid commented:
i actually got my hands on a windows laptop to look at this script with ^_^

4strid commented:
so i gave your branch a quick SPINNY SPINNY and I think I am going to take another looksie at some of your changes that may not have been necessary . I think the popup was fine how it was before . i can't get the WASD keys to work ? among other things . I have to go to work now but i'll poke around some more when I get back

Ulyanov-programmer commented:
> i can't get the WASD keys to work 

This was exactly the problem - there was a separate mode for HJKL, where only the HJKL keys worked, but there was no such mode for WASD (or rather, it was, but it used the faulty Win + G hotkey).
I reworked it so that Shift + Space would switch specific layout options - left Shift + Space: WASD mode, right Shift + Space: HJKL mode.

4strid commented:
I would like to just always have both sets of keys be active at all times . Maybe it made sense to give people options (maybe, vim keys are *scary* 😱!) but like , it doesn't , really have to be that way ?

Ulyanov-programmer commented:
> always have both sets of keys be active at all times

You seem to live in the USA, not in India, and such brilliant UX ideas that create many collisions in hot keys should not occur to you. I don't see the point in this, if this is a matter of principle for you - I will not seek a compromise and will leave the project in your hands.

I should clarify – it turns out that the only issue on which we do not agree is this?

4strid commented:
Welllll basically , I'd like it to work pretty close to how it used to ? Which it just presently doesn't ..
